### DISCUSSION TITLE
Is it possible to conditionally include a *-to-one relation?

### DISCUSSION THREAD

**Discussion by ymc9** [prisma/prisma#15723] [Q&A] [answered]
Hi there,

I understand that when using "include" for "one-to-many" relation, I can use a "where" clause to filter the "many" side. However, it seems not possible to do that for "one-to-one" or "many-to-one". My scenario is that I only want to include a relation when it meets some condition, otherwise it should be null.

Example schema:
```prisma
model List {
  id String @id
  todos Todo[]
  private Boolean
}

model Todo {
  id String @id
  title String
  list List @relation(fields:[listId], references: [id])
  listId String
}
```

The kind of query I hope would be supported:
```ts
prisma.todo.findMany({
  include: {
    list: {
        private: false
    }
  }
});
```

Is there an underlying reason not to allow this?

**Comment by pantharshit00** [comment, marked as answer]
Hello,

Looks like this is currently not possible. I would suggest you to open a feature request here: https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=


The workaround here would be to filter the data in the application after you have fetched it.

**Reply by vladi-stevanovic** [reply]
Hello @ymc9 - did you perhaps have a chance to try the workaround recommended by Harshit? Let us know if you ended up opening a feature request!

**Reply by ymc9** [reply]
Yes, I can do post-process filtering, it's a bit more complicated than directly embedding the condition in a query. I'm working on adding data access policies above the Prisma schema so need to handle the check in a generic way.

I'll open a feature request. Thank you, guys.

**Comment by antja03** [comment]
I'm not sure if I'm understanding this correctly, but if you're trying to filter for todos where the list is not private, you could use this query:
```ts
prisma.todos.findMany({
    where: {
        list: {
            private: false
        }
    }
});
```

If you're trying to exclude the private column from list in all fetched todos, you could use this query:
```ts
prisma.todos.findMany({
    include: {
        list: {
            include: {
                private: false
            }
        }
    }
})
```

**Reply by ymc9** [reply]
Hi, my requirement is a bit strange. I want to always get the Todo object, but conditionally fetch it’s related List object based on filter over List.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
